dataprofiler/profilers/profile_builder.py

```diff
@@ -230,7 +236,37 @@ def _update_base_stats(self, base_stats):
             self.null_types, list(base_stats["null_types"].keys())
         )
 
-        for null_type, null_rows in base_stats["null_types"].items():
+        base_min = base_stats["min_id"]
+        base_max = base_stats["max_id"]
+        base_nti = base_stats["null_types"]
+
+        # Check if indices overlap, if they do, adjust attributes accordingly
+        if utils.overlap(self.min_id, self.max_id, base_min, base_max):
+            warnings.warn(f"Overlapping indices detected. To resolve, indices "
+                          f"where null data present will be shifted forward "
+                          f"when stored in profile: {self.name}")
+
+            # Shift indices (min, max, and all indices in null types index
+            base_min = base_min + self.max_id + 1
+            base_max = base_max + self.max_id + 1
+
+            base_nti = {k: {x + self.max_id + 1 for x in v} for k, v in
+                        base_stats["null_types"].items()}
+
+        # Store/compare min/max id with current
+        if isinstance(base_min, int):
+            if not isinstance(self.min_id, int):
+                self.min_id = base_min
+            else:
+                self.min_id = min(self.min_id, base_min)
+        if isinstance(base_max, int):
+            if not isinstance(self.max_id, int):
+                self.max_id = base_max
+            else:
+                self.max_id = max(self.max_id, base_max)
+
+        # Update null row indices
+        for null_type, null_rows in base_nti.items():
```

Comment:
@lettergram @JGSweets consolidated the overlap detection/index shift logic to be applied only to the null_type_index (not the index in data destructively) during _update_base_stats, so now this one solution covers both of the previous cases' scopes.